change/@ni-nimble-components-e1eca008-24b5-481f-a835-1e3157d4ac1b.json

```diff
@@ -0,0 +1,7 @@
+{
```

Comment:
@aagash-ni There were several things I think we should update to make the update codepath clearer and some refactors. I captured them here: https://github.com/ni/nimble/pull/1699/files

I didn't figure out yet what is causing the test failures but I think it's a direction worth pursuing. Could you see if we could merge those changes in this PR and if you have ideas of why the tests are failing?

Reply:
Updated the changes.